projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -2998,6 +2998,7 @@ export class IgxGridBaseDirective extends DisplayDensityBase implements
      * @internal
      */
     public setFilteredSortedData(data, pinned: boolean) {
+        if (data === undefined) { data = []; }
```

Comment:
Could be shortened: 
data = data ? data.map(rec => rec.ghostRecord !== undefined ? rec.recordRef : rec) : [];